feat: Add icons migration (#4108)
Workday/canvas-kit: modules/docs/llm/icons/icon-migration-codemod.md

> # Migration codemod

Reviewer: ```suggestion
# Migration Codemod
```

The **`icon-migration`** codemod reads deprecation metadata shipped with

> `@workday/canvas-system-icons-web` and updates imports and usages in your source files

Reviewer: ```suggestion
`@workday/canvas-system-icons-web` and updates imports and usage instances in your source files
```

automatically. For what changed in v5 and package installation, see the
[Sana Canvas Assets overview](/docs/guides-icon-migration-sana-canvas-assets-overview--docs).

For general advice on running Canvas Kit codemods (review process, formatting, file types), see
[Codemods](/docs/guides-codemods--docs).

## Table of Contents

- [Running the codemod](#running-the-codemod)
- [What the codemod does](#what-the-codemod-does)
- [All icon transforms](#all-icon-transforms)
  - [v4 naming convention migrations](#v4-naming-convention-migrations)
  - [v5 Sana Canvas Assets changes](#v5-sana-canvas-assets-changes)
- [Example transformations](#example-transformations)
  - [Named import](#named-import)
  - [Import alias preserved](#import-alias-preserved)
  - [Fallback already imported](#fallback-already-imported)
- [Manual migration and lookup](#manual-migration-and-lookup)
- [What the codemod does not do](#what-the-codemod-does-not-do)
- [After migration](#after-migration)

> ## Running the codemod

Reviewer: ```suggestion
## Running the Codemod
```

yarn remove @workday/canvas-kit-codemod
```

Provide `[path]` to the folders that contain your application source (for example `src/` or specific
packages in a monorepo). Limiting scope speeds up the run and reduces unrelated diffs.

> **Note:** The codemod only transforms `.js`, `.jsx`, `.ts`, and `.tsx` files. Update other file
> types (`.json`, `.mdx`, `.md`, Storybook stories in non-TS formats, etc.) manually. You may need
> to run your linter after the codemod. Output formatting may not match your project style.

**Tips before migrating:**

- **Run in batches.** Prefer a focused directory (for example `src/components`) instead of the
  entire monorepo. Smaller diffs are easier to review and revert.
- **Review every change.** Codemods are not bulletproof. Verify icons in the UI, especially where
  semantics matter (status, navigation, or custom branding).
- **Expressive, accent, and applet icons.** The `icon-migration` codemod only touches
  `@workday/canvas-system-icons-web`. Use the `v15-icons` transform for accent, applet, and
  expressive icons, covered in the
  [v15 upgrade guide](/docs/guides-upgrade-guides-v15-0-overview--docs#codemod-transformations-for-icons).

> ## What the codemod does

Reviewer: ```suggestion
## What the Codemod Does
```
